plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,95 @@
 #!/usr/bin/env python
 import re
 import sys
+import json
+import argparse
 from subprocess import CalledProcessError
+from collections import Counter
 
-from openshift import oc
+import openshift
 import nagios
 
 
-check_disk_cmd = ("df", "--output=target,pcent,ipcent")
+check_disk_cmd = ("df", "--output=pcent,ipcent,target")
 # Example output:
 # /etc/hosts      19%    8%
-check_disk_output_pattern = re.compile(r"([\w/]+)\s+(\d+)%\s+(\d+)%")
+check_disk_output_pattern = re.compile(r"\s*(\d+)%\s+(\d+)%\s+(.+)$")
 
 
-def parse_df(line):
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-w", "--warn", type=int, action="store",
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
-    return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
+    return mo.group(3), int(mo.group(1)), int(mo.group(2))
+
+
+def parse_df_lines(lines):
+    return filter(None, map(parse_df_line, lines.splitlines()))
+
+
+def analize(pod, disks, warning_threshold, critical_threshold):
```

Comment:
I haven't realized until looking further at the other changes.

Before, `analize` was focused on the "business logic": it takes just 4 numbers that it needs in order to tell if things are fine or not. It used to return two booleans answering two questions: "are we above warning level?", and "are we above critical level?".

Analyzing a container (a pod may have multiple containers, not sure if we're taking that into account) is a matter of analyzing with `analize` every line, and checking whether any of those booleans are true.

At this point I'm afraid I'm inflicting too much of a "personal preference" and sticking into the old code base into the review.

This PR adds a lot more code than I would expect, thinking that we were only missing the reporting part + bug fixes?